triggers/runner/src/main/scala/com/daml/trigger/Runner.scala

```diff
@@ -109,20 +106,48 @@ class Runner(
         newState
       }
       case v => {
-        throw new RuntimeException(s"Expected Tuple3 but got $v")
+        throw new RuntimeException(s"Expected Tuple2 but got $v")
       }
     }
 
+  def logTraces(machine: Speedy.Machine) = {
+    machine.traceLog.iterator.foreach {
+      case (msg, optLoc) =>
+        logger.info(s"TRACE ${Pretty.prettyLoc(optLoc).render(80)}: $msg")
+    }
+  }
+
+  def stepToValue(machine: Speedy.Machine) = {
+    while (!machine.isFinal) {
+      machine.step() match {
+        case SResultContinue => ()
```

Comment:
Does `machine.traceLog` accumulate log messages across `step`s, or do we need to `logTraces` here too, to avoid swallowing traces?

Reply:
It accumulates them between steps. There is currently a limit of `100` but that seems high enough that I think it’s fine for now to only do it once after the end.